python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -8170,58 +8170,99 @@ def test_agg_for_dataframe_with_string_columns(aggs):
+        {"g": [2.0, np.nan, 4.0], "n": [np.nan, np.nan, np.nan]},
+        {"d": [np.nan, np.nan, np.nan], "e": [np.nan, np.nan, np.nan]},
+        {"a": [1.0, 2, 3], "b": pd.Series([4.0, 8.0, 3.0], index=[1, 2, 3])},
+        {
+            "d": [1.0, 2.0, 3.0],
+            "c": pd.Series([np.nan, np.nan, np.nan], index=[1, 2, 3]),
+        },
+        {
+            "a": [False, True, False],
+            "b": pd.Series([1.0, 2.0, np.nan], index=[1, 2, 3]),
+        },
+        {
+            "a": [np.nan, np.nan, np.nan],
+            "e": pd.Series([np.nan, np.nan, np.nan], index=[1, 2, 3]),
+        },
     ],
 )
-def test_dataframe_roundtrip_arrow_list_dtype(gdf):
-    table = gdf.to_arrow()
-    expected = gd.DataFrame.from_arrow(table)
+@pytest.mark.parametrize(
+    "data2",
+    [
+        {"b": [3, 5, 6], "e": [8, 2, 1]},
+        {"c": [True, False, True], "d": [3.0, 4.0, 5.0]},
+        {"e": [False, False, True], "g": [True, True, False]},
+        {"g": [np.nan, np.nan, np.nan], "c": [np.nan, np.nan, np.nan]},
+        {"a": [7, 5, 8], "b": pd.Series([2.0, 7.0, 9.0], index=[0, 1, 2])},
+        {
+            "b": [np.nan, 2.0, np.nan],
+            "c": pd.Series([2, np.nan, 5.0], index=[2, 3, 4]),
+        },
+        {
+            "a": [True, np.nan, True],
+            "d": pd.Series([False, True, np.nan], index=[0, 1, 3]),
+        },
+    ],
+)
+def test_update_for_dataframes(
+    data, data2, join, overwrite, filter_func, errors
+):
+    pdf = pd.DataFrame(data)
+    gdf = gd.DataFrame(data)
+
+    other_pd = pd.DataFrame(data2)
+    other_gd = gd.DataFrame(data2)
 
-    assert_eq(gdf, expected)
+    expect = pdf.update(other_pd, join, overwrite, filter_func, errors)
+    got = gdf.update(other_gd, join, overwrite, filter_func, errors)
+
+    assert_eq(expect, got)
 
 
 @pytest.mark.parametrize(
-    "gdf",
-    [
-        gd.DataFrame({"a": [{"one": 3, "two": 4, "three": 10}]}),
-        gd.DataFrame(
-            {
-                "left-a": [0, 1, 2],
-                "a": [{"x": 0.23, "y": 43}, None, {"x": 23.9, "y": 4.3}],
-                "right-a": ["abc", "def", "ghi"],
-            }
-        ),
-        gd.DataFrame(
-            {
-                "left-a": [{"a": 1}, None, None],
-                "a": [
-                    {"one": 324, "two": 23432, "three": 324},
-                    None,
-                    {"one": 3.24, "two": 1, "three": 324},
-                ],
-                "right-a": ["abc", "def", "ghi"],
-            }
-        ),
-    ],
+    "join", ["right"],
+)
+def test_update_for_right_join(join):
+    gdf = gd.DataFrame({"a": [1, 2, 3], "b": [3.0, 4.0, 5.0]})
+    other_gd = gd.DataFrame({"a": [1, np.nan, 3], "b": [np.nan, 2.0, 5.0]})
+
+    with pytest.raises(
+        NotImplementedError, match="Only left join is supported"
+    ):
+        gdf.update(other_gd, join)
+
+
+@pytest.mark.parametrize(
+    "errors", ["raise"],
 )
-def test_dataframe_roundtrip_arrow_struct_dtype(gdf):
```

Comment:
I think `test_dataframe_roundtrip_arrow_struct_dtype` & `test_dataframe_roundtrip_arrow_list_dtype` are deleted accidentally while resolving a merge conflict, we must add them back

Reply:
oh yikes, how do i recover them?  the two tests weren't written by me :((

Reply:
Can be copy-pasted from `branch-0.18` version of the file.